src/server/pfs/server/api_server.go

```diff
@@ -194,14 +187,27 @@ func (a *apiServer) FinishCommit(ctx context.Context, request *pfs.FinishCommitR
 	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
 	if err := a.txnEnv.WithTransaction(ctx, func(txn txnenv.Transaction) error {
 		return txn.FinishCommit(request)
+	}, func(txnCtx *txncontext.TransactionContext) (string, error) {
+		// FinishCommit in a transaction by itself has special handling with regards
```

Comment:
How is this handled if you finish a commit in a transaction with other operations? It is not clear to me what the commitset ID should be in that case. Also, would it be possible to handle this through an alias for the finished commit so that we don't need to do this override thing, which seems to exist because we haven't found a good way to incorporate triggers / deferred processing into the global ID model.

Reply:
> How is this handled if you finish a commit in a transaction with other operations?
If there is a user-level transaction, this will use the transaction ID.

> would it be possible to handle this through an alias for the finished commit
Yes, that is what this does, it's just that aliases are written to always use the transaction ID, so this overrides the transaction ID.